src/libmeasurement_kit/nettests/multi_ndt.cpp

```diff
@@ -0,0 +1,241 @@
+
+    try {
+        // Note: do static case to make sure it's convertible to a double
+        return static_cast<double>(test_s2c["web100_data"]["MinRTT"]);
+    } catch (const std::exception &) {
+        logger->warn("Cannot access Web100 data");
+        /* Fallthrough to next method of computing RTT */
+    }
+
+    std::vector<double> rtts;
+    try {
+        // XXX Needed to add temp variable because otherwise it did not compile
+        std::vector<double> temp = test_s2c["connect_times"];
+        rtts = temp;
+    } catch (const std::exception &) {
+        logger->warn("Cannot access connect times");
+        /* Fallthrough to the following check that will fail */ ;
+    }
+    if (rtts.size() <= 0) {
+        logger->warn("Did not find any reliable way to compute RTT");
+        return nullptr; /* We cannot compute the RTT */
+    }
+
+    double sum = 0.0;
+    for (auto &rtt: rtts) {
+        sum += rtt * 1000.0 /* To milliseconds! */;
+    }
+    return sum / rtts.size();  /* Division by zero excluded above */
+}
+
+static report::Entry compute_download_speed(report::Entry &test_s2c,
+                                            Var<Logger> logger) {
+    try {
+        std::vector<double> speeds;
+        for (auto &x: test_s2c["receiver_data"]) {
+            speeds.push_back(x[1]);
+        }
+        std::sort(speeds.begin(), speeds.end());
+        std::vector<double> good_speeds(
+            // Note: going beyond vector limits would raise
+            // a std::length_error exception
+            speeds.begin() + 6, speeds.end() - 2
+        );
+        double sum = 0.0;
+        for (auto &x : good_speeds) {
+            sum += x;
+        };
+        if (good_speeds.size() <= 0) {
+            logger->warn("The vector of good speeds is empty");
+            return nullptr;
+        }
+        return sum / good_speeds.size();
+    } catch (const std::exception &) {
+        logger->warn("Cannot compute download speed");
+        // FALLTHROUGH
+    }
+    return nullptr;
+}
+
+static report::Entry compute_stats(report::Entry &root, std::string key,
+                                   Var<Logger> logger) {
+    report::Entry stats;
+    report::Entry test_s2c;
+    try {
+        test_s2c = root[key]["test_s2c"][0] /* We know it's just one entry */;
+    } catch (const std::exception &) {
+        logger->warn("cannot access root[\"%s\"][\"test_s2c\"]", key.c_str());
+        return nullptr;  /* Cannot compute this stat */
+    }
+    stats["ping"] = compute_ping(test_s2c, logger);
+    stats["download"] = compute_download_speed(test_s2c, logger);
+    stats["key"] = key;
+    return stats;
+}
+
+static void compute_simple_stats(report::Entry &entry, Var<Logger> logger) {
+    report::Entry single = compute_stats(entry, "single_stream", logger);
+    report::Entry multi = compute_stats(entry, "multi_stream", logger);
+    report::Entry selected;
+
```

Comment:
It would be nice to have a comment here explaining that you pick the statistics that have the highest download speed for the simple stats.